crates/solver/src/settlement_ranker.rs

```diff
@@ -194,6 +195,27 @@ impl SettlementRanker {
             );
         }
 
+        // Filter out settlements with non-positive score.
+        if Utc::now() > self.auction_rewards_activation_timestamp // CIP20 activated
+        && self.skip_non_positive_score_settlements
```

Comment:
Not sure if it makes sense to have `CIP-20` code enabled while also allowing negative scores. If this is not the case we don't need the new flag. I guess this gives us more flexibility so we don't have to decide this matter right now (we could simply always have this flag be true).
I guess I'm saying that to me it seems like we should not have this flag as allowing negative scores could introduce problems but I'm not 100% sure.

Reply:
Essentially, we are afraid that there could be a lot of consecutive auctions where winner would have negative objective value and therefore skipping the submission part, so we need a fast way to allow negative scores after all. 
The whole `if` will be dropped anyway in https://github.com/cowprotocol/services/issues/1352

Reply:
I get the idea. I was just not sure how compatible that is with the solver rewards computation. Would it maybe make sense to use `score == 0, reference_score == 0` for the reward computation in those cases?

Reply:
So, indeed, we don't want to allow negative scores, in the sense that the trivial solution with score zero would always beat them. And explicitly not submitting the empty solution and allowing negative scores can lead to problems, the most obvious being an auction with only one submitted solution of negative score; what is the reference score in that case? Also, negative scores can result in aggressive gaming under certain cases.

Long story short, allowing negative scores, which should only happen temporarily, and while ensuring that solvers report truthfully  it is really a last-resort solution, in case we see that we are filtering out tons of auctions. But, again, the plan is not to activate this flag at all.

Reply:
If we fallback to allow negative scores, it just means that those with negative objective values will have score 0 and considered as 0 in ranking. But as @harisang  said, we expect this to not happen at all.